pkg/cmd/experimental/router/router.go

```diff
@@ -144,6 +145,7 @@ func RunCmdRouter(f *clientcmd.Factory, cmd *cobra.Command, out io.Writer, cfg *
 			return cmdutil.UsageError(cmd, "You may not pass negative labels in %q", cfg.Labels)
 		}
 		label = valid
+		nodeSelector = label
```

Comment:
Is this always true?  What if I want a node selector that doesn't match what I'm labeling my router with?  This is probably best added as a new, optional flag.